core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/conditional/DecodeExpression.kt

```diff
@@ -32,11 +32,35 @@ class DecodeExpression<T : ValidType, U : ValidType>(
                 } + defaultDopeQuery?.parameters.orEmpty(),
         )
     }
+
+    private fun getSearchResultDopeQuery(searchResult: SearchResult<T, out U>): DopeQuery {
+        val searchDopeQuery = searchResult.searchExpression.toDopeQuery()
+        val resultDopeQuery = searchResult.resultExpression.toDopeQuery()
+        return DopeQuery(
+            queryString = "${searchDopeQuery.queryString}, ${resultDopeQuery.queryString}",
+            parameters = searchDopeQuery.parameters + resultDopeQuery.parameters,
+        )
+    }
 }
 
+@JvmName("decodeWithGeneric")
 fun <T : ValidType, U : ValidType> decode(
     decodeExpression: UnaliasedExpression<T>,
     searchResult: SearchResult<T, U>,
     vararg searchResults: SearchResult<T, U>,
-    default: UnaliasedExpression<U>? = null,
+) = DecodeExpression(decodeExpression, searchResult, *searchResults)
+
+fun <T : ValidType, U : ValidType> decode(
+    decodeExpression: UnaliasedExpression<T>,
+    searchResult: SearchResult<T, U>,
+    vararg searchResults: SearchResult<T, U>,
+    default: UnaliasedExpression<U>,
 ) = DecodeExpression(decodeExpression, searchResult, *searchResults, default = default)
+
+@JvmName("decodeWithoutGeneric")
+fun <T : ValidType> decode(
+    decodeExpression: UnaliasedExpression<T>,
+    searchResult: SearchResult<T, out ValidType>,
+    vararg searchResults: SearchResult<T, out ValidType>,
+    default: UnaliasedExpression<out ValidType>? = null,
+) = DecodeExpression<T, ValidType>(decodeExpression, searchResult, *searchResults, default = default)
```

Comment:
You do not have to explicitly state the `T, ValidType` generic

Reply:
done